nav2_util/include/nav2_util/node_utils.hpp

```diff
@@ -93,6 +93,20 @@ void declare_parameter_if_not_declared(
   }
 }
 
+template<typename NodeT>
+std::string get_plugin_type_param(
+  NodeT node,
+  const std::string & plugin_name)
+{
+  declare_parameter_if_not_declared(node, plugin_name + ".plugin");
+  std::string plugin_type;
+  if (!node->get_parameter(plugin_name + ".plugin", plugin_type)) {
```

Comment:
This is wrong then - you need to pass it the plugin namespace

Reply:
I'm not sure if this is correct.

```yaml
node_name:
  ros__parameters:
    plugins: ["plugin_1", "plugin_2"]
    plugin_1:
      plugin: "Plugin1/Type"
    plugin_2:
      plugin: "Plugin2/Type"
```

If this is how the parameters are passed, we would get the plugin type this way:
```cpp
node->get_parameter("plugin_1" + ".plugin", plugin_type);
```

Are you suggesting that we place the plugin specific parameters under "plugins" (or "controller_plugins" etc.) namespace? Something like this:
```yaml
node_name:
  ros__parameters:
    plugins: ["plugin_1", "plugin_2"]
      plugin_1:
        plugin: "Plugin1/Type"
      plugin_2:
        plugin: "Plugin2/Type"
```

Reply:
But you just changed `plugins` to `recovery_plugins`, `controller_plugins`, and `planner_plugins`

Reply:
Yup, but `recovery_plugins`, `controller_plugins`, and `planner_plugins` are only needed to get the plugin names. The `plugin` parameter for plugin type is available under each individual plugin name's namespace so it is not dependent on `recovery_plugins`, `controller_plugins`, and `planner_plugins`.

Reply:
Got it, makes sense